omniduct/filesystems/s3.py

```diff
@@ -0,0 +1,154 @@
+import os
+import re
+
+import pandas as pd
+
+from .base import FileSystemClient
+from ..utils.debug import logger
+
+
+class S3Client(FileSystemClient):
+
+    PROTOCOLS = ['s3']
+    DEFAULT_PORT = 80
+
+    def _init(self, bucket=None, aws_profile=None):
+        assert bucket is not None, 'S3 Bucket must be specified using the `bucket` kwarg.'
+        self.bucket = bucket
+        self.aws_profile = aws_profile
+
+        self._setup_session()
```

Comment:
why call setup_session here when it's also called in _connect?

Reply:
We need the right host end-point to be used when checking whether a `Duct` is available; and that end-point depends  on the aws region; which is most easily attained by configuring the client using (perhaps stale) credentials. If those credentials are not stale, the `_connect()` method won't actually be run. Add a comment to clarify this.